### PR TITLE
AK+Kernel+Tests: Change EnumBits has_flag() to check all flags in mask are present

### PR DESCRIPTION
**Pull request by timmot**
I noticed this when I saw `has_flags(Core::OpenMode::ReadOnly, Core::OpenMode::ReadOnly | Core::OpenMode::WriteOnly)` was returning true.

My expectation would be that this fails.

### CODE REVIEW

**Review by bgianfo — APPROVED** [review]
Nice catch, thanks for fixing this!

**Comment by bgianfo** [comment]
As discussed on discord, it looks like this now fails the test: 

```
$  ./Meta/Lagom/TestEnumBits_lagom
...
Running test 'has_flag'.
FAIL: ../../Tests/AK/TestEnumBits.cpp:67: EXPECT(has_flag(intro, VideoIntro::Friends)) failed
FAIL: ../../Tests/AK/TestEnumBits.cpp:68: EXPECT(!has_flag(intro, VideoIntro::Well)) failed
FAIL: ../../Tests/AK/TestEnumBits.cpp:69: EXPECT(!has_flag(intro, VideoIntro::CompleteIntro)) failed
Failed test 'has_flag' in 0ms
Finished 8 tests and 0 benchmarks in 0ms (0ms tests, 0ms benchmarks, 0ms other).
Out of 8 tests, 7 passed and 1 failed.
```

That's probably why the state machine generator is now failing in the build 😄

File: AK/EnumBits.h [hunk 1]
```diff
@@ -20,62 +20,62 @@
...
+        using Type = UnderlyingType<Enum>;                                 \
+        lhs = static_cast<Enum>(                                           \
+            static_cast<Type>(lhs) ^ static_cast<Type>(rhs));              \
+        return lhs;                                                        \
+    }                                                                      \
+                                                                           \
+    Prefix constexpr bool has_flag(Enum value, Enum mask)                  \
+    {                                                                      \
+        using Type = UnderlyingType<Enum>;                                 \
+        return static_cast<Type>(value & mask) != static_cast<Type>(mask); \
```
Comment (bgianfo) [inline AK/EnumBits.h:80]: As discussed on discord, this should be: 

```suggestion
        return static_cast<Type>(value & mask) == static_cast<Type>(mask); \
```

**Review by bgianfo — CHANGES_REQUESTED** [review]

**Comment by timmot** [comment]
It appears that more things are dependent on the previous functionality, on boot we get the grey wallpaper and mouse but no windows, icons, or the taskbar.

**Review by kleinesfilmroellchen — APPROVED** [review]

**Comment by awesomekling** [comment]
The commit messages are very confusing, they seem to be describing a problem(?)
"EnumBits has_flag should require all flags present" doesn't tell us what the commit is actually doing.

**Comment by timmot** [comment]
I have updated the commit messages, let me know if you'd like more clarity.

**Pull request merged**